promslog/slog.go

```diff
@@ -155,55 +70,60 @@ func (l *AllowedLevel) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if s == "" {
 		return nil
 	}
-	lo := &AllowedLevel{}
-	if err := lo.Set(s); err != nil {
+	if err := l.Set(s); err != nil {
 		return err
 	}
-	*l = *lo
 	return nil
 }
 
-func (l *AllowedLevel) String() string {
-	return l.s
-}
-
-// Set updates the value of the allowed level.
-func (l *AllowedLevel) Set(s string) error {
-	if l.lvl == nil {
-		l.lvl = &slog.LevelVar{}
+// String returns the current level.
+func (l *Level) String() string {
+	switch l.lvl.Level() {
+	case slog.LevelDebug:
+		return "debug"
+	case slog.LevelInfo:
+		return "info"
+	case slog.LevelWarn:
+		return "warn"
+	case slog.LevelError:
+		return "error"
+	default:
+		return ""
 	}
+}
 
+// Set updates the logging level with the validation.
+func (l *Level) Set(s string) error {
 	switch strings.ToLower(s) {
 	case "debug":
 		l.lvl.Set(slog.LevelDebug)
-		callerAddFunc = true
 	case "info":
 		l.lvl.Set(slog.LevelInfo)
-		callerAddFunc = false
 	case "warn":
 		l.lvl.Set(slog.LevelWarn)
-		callerAddFunc = false
 	case "error":
 		l.lvl.Set(slog.LevelError)
-		callerAddFunc = false
 	default:
 		return fmt.Errorf("unrecognized log level %s", s)
 	}
-	l.s = s
 	return nil
 }
 
-// AllowedFormat is a settable identifier for the output format that the logger can have.
-type AllowedFormat struct {
+// Format controls a logging output format.
+// Not concurrency-safe.
+type Format struct {
```

Comment:
As mentioned in the description - this is a breaking change, let me know if this is ok.

Reply:
Yes, this is a breaking change.

Any chance you would keep the old struct around and add a `Deprecated` comment so downstream Go users get a linter warning?

Reply:
Or, maybe it's better to just drop the struct so it's a compile error.

Reply:
For deprecated flow we would need to maintain a behaviour which means `promslog.Config` using old thing, we can but until we are v1 and this is not too painful we can break.